R/frailty_score.R

```diff
@@ -1,152 +1,174 @@
+#'
+#' A warning message is returned if `erdiag` is explicitly set to `NULL`.
+#' It's possible to exclude diagnoses at the emergency department from the consideration by explicitly specifying `erdiag=NULL`.
+#' However, excluding any types of diagnoses is not recommended.
+#' The CIHI frailty score was developed and validated based on all diagnoses present in medical records (both NACRS and DAD).
+#' Excluding diagnoses in NACRS was found to underestimate frailty levels (Amuah et al, 2023).
+#'
+#' @section Notes:
+#' The previous `frailty_score()` function calculates the UK HFRS (Gilbert, 2018), and it is now deprecated.
+#' Using a similar approach as the UK HFRS, the CIHI HFRS was developed and validated based on Canadian cohorts, making it particularly suited for GEMINI data.
+#' The UK version remains available in `Rgemini` version 0.3.1 and earlier but will not receive future maintenance.
+#' Users interested in the UK version should refer to the original publications for important differences in diagnostic coding practices and age threshold.
 #'
 #' @references
-#' Gilbert T, Neuburger J, Kraindler J, et al. Development and validation of a
-#' Hospital Frailty Risk Score focusing on older people in acute care settings using
-#' electronic hospital records: an observational study. Lancet 2018; published online
-#' April 26. http://dx.doi.org/10.1016/S0140-6736(18)30668-8.
+#' UK HFRS: Gilbert T, et al. Lancet, 2018. http://dx.doi.org/10.1016/S0140-6736(18)30668-8
+#' CIHI HFRS: Amuah JE, et al. CMAJ, 2023. https://doi.org/10.1503/cmaj.220926
+#' CIHI methodology notes: https://www.cihi.ca/sites/default/files/document/cihi-hospital-frailty-risk-measure-meth-notes-en.pdf
+#' We recommend referencing both original articles (UK HFRS and CIHI HFRS) when using this function.
 #'
+#' @examples
+#' \dontrun{
+#' cohort_dum <- data.table(genc_id=c(1, 2, 3), age=c(64, 65, 80))
+#' ipdiag_dum <- dummy_diag(nid=3, nrow=10, ipdiagnosis=TRUE, pattern = "C20$|R460$") # frailty conditions
+#' erdiag_dum <- dummy_diag(nid=3, nrow=5, ipdiagnosis=FALSE, pattern = "M121$") # not a frailty condition
+#' frailty_score(cohort_dum, ipdiag_dum, erdiag_dum, component_wise = FALSE)
+#' }
+#'
+#' @importFrom fuzzyjoin regex_left_join 
+#' 
 #' @export
-frailty_score <- function(cohort,
-                          ipdiagnosis,
-                          eddiagnosis) {
 
-  ## Ensure user inputs are in data.table format before proceeding
+frailty_score  <- function(cohort, ipdiag, erdiag, component_wise = FALSE) {
+
+  # load CIHI HFRS-ICD mapping from package data folder
+  data("mapping_cihi_hfrs_icd", package = "Rgemini")
+  frailty_map <- mapping_cihi_hfrs_icd %>% mutate(diagnosis_code = gsub("\\.", "", icd10ca)) %>% data.table()
+
+  # input checks
+  check_input(cohort, c("data.table", "data.frame"), 
+              colnames = c("genc_id", "age"),
+              coltypes = c("", "numeric"))
```

Comment:
Oh, one more thing: `class(admdad$age)` actually returns `"integer"` so this will result in an error unless users change class of age to "numeric". You can specify it like this to allow both numeric and integer type for age: 
```
coltypes = c("", "numeric|integer")
```

Reply:
Thank you for the review @loffleraSMH! I made the minor edits to the documentation and file name you suggested and fixed the input check for age class. I will proceed to merge. (currently ci\cd check fails due to a documentation issue on `develop` branch, will merge once that's resolved).